drivers/gpio/gpio_nrfx.c

```diff
@@ -0,0 +1,467 @@
+	struct gpio_nrfx_data *data = get_port_data(port);
+
+	u32_t port_in = nrf_gpio_port_in_read(reg) ^ data->inverted;
+
+	if (access_op == GPIO_ACCESS_BY_PORT) {
+		*value = port_in;
+	} else {
+		*value = (port_in & BIT(pin)) ? 1 : 0;
+	}
+
+	return 0;
+}
+
+static int gpio_nrfx_manage_callback(struct device *port,
+				     struct gpio_callback *callback,
+				     bool set)
+{
+	_gpio_manage_callback(&get_port_data(port)->callbacks, callback, set);
+	return 0;
+}
+
+static int gpio_nrfx_pin_manage_callback(struct device *port,
+					 int access_op,
+					 u32_t pin,
+					 bool enable)
+{
+	struct gpio_nrfx_data *data = get_port_data(port);
+	int res = 0;
+	u8_t from_pin;
+	u8_t to_pin;
+
+	if (access_op == GPIO_ACCESS_BY_PORT) {
+		from_pin = 0;
+		to_pin   = 31;
+	} else {
+		from_pin = pin;
+		to_pin   = pin;
+	}
+
+	for (u32_t curr_pin = from_pin; curr_pin <= to_pin; ++curr_pin) {
+		WRITE_BIT(data->int_en, curr_pin, enable);
+
+		res = gpiote_pin_int_cfg(port, curr_pin);
+		if (res != 0) {
+			return res;
+		}
+	}
+
+	return res;
+}
+
+static inline int gpio_nrfx_pin_enable_callback(struct device *port,
+						int access_op,
+						u32_t pin)
+{
+	return gpio_nrfx_pin_manage_callback(port, access_op, pin, true);
+}
+
+static inline int gpio_nrfx_pin_disable_callback(struct device *port,
+						 int access_op,
+						 u32_t pin)
+{
+	return gpio_nrfx_pin_manage_callback(port, access_op, pin, false);
+}
+
+static const struct gpio_driver_api gpio_nrfx_drv_api_funcs = {
+	.config = gpio_nrfx_config,
+	.write = gpio_nrfx_write,
+	.read = gpio_nrfx_read,
+	.manage_callback = gpio_nrfx_manage_callback,
+	.enable_callback = gpio_nrfx_pin_enable_callback,
+	.disable_callback = gpio_nrfx_pin_disable_callback
+};
+
+/**
+ * @brief Function for getting pins that triggered level interrupt.
+ *
+ * @param port Pointer to GPIO port device.
+ *
+ * @return Bitmask where 1 marks pin as trigger source.
+ */
+static u32_t check_level_trigger_pins(struct device *port)
+{
+	if (port == NULL) {
+		return 0;
+	}
+	struct gpio_nrfx_data *data = get_port_data(port);
+
+	u32_t port_in = nrf_gpio_port_in_read(get_port_cfg(port)->port);
+
+	/* Extract which pins after inversion, have logic level same as
+	 * interrupt trigger level.
+	 */
+	u32_t pin_states = ~(port_in ^ data->inverted ^ data->active_level);
+
+	/* Take into consideration only pins that were configured to
+	 * trigger interrupts and have callback enabled.
+	 */
+	u32_t out = pin_states & data->int_en & data->pin_int_en;
+
+	/* Exclude pins that trigger interrupts by edge. */
+	out &= ~data->trig_edge & ~data->double_edge;
+
+	return out;
+}
+
+static inline void fire_callbacks(struct device *port, u32_t pins)
+{
+	_gpio_fire_callbacks(&get_port_data(port)->callbacks, port, pins);
+}
+
+#ifdef CONFIG_GPIO_NRFX_P0
+DEVICE_DECLARE(gpio_nrfx_p0);
+#endif
+#ifdef CONFIG_GPIO_NRFX_P1
+DEVICE_DECLARE(gpio_nrfx_p1);
+#endif
+
+static void gpiote_event_handler(void)
+{
+	u32_t int_trigger_pins[GPIO_COUNT];
```

Comment:
This should be zeroed here. And the calls to `check_level_trigger_pins` below should be done only when the PORT event is set as there is no need to call them otherwise. Besides, this way you hide some logic here and it may generate confusion (and errors) when some modifications are needed here in some future.

And what about renaming it to `fired_triggers`?